src/Microsoft.ML.Data/Transforms/ConcatEstimator.cs

```diff
@@ -0,0 +1,424 @@
+            bool hasSlotNames = false;
+
+            // We will get the item type from the first column.
+            ColumnType itemType = null;
+
+            for (int i = 0; i < sources.Length; ++i)
+            {
+                if (!inputSchema.TryFindColumn(sources[i], out var col))
+                    throw _host.ExceptSchemaMismatch(nameof(inputSchema), "input", sources[i]);
+                if (i == 0)
+                    itemType = col.ItemType;
+                // For the sake of an estimator I am going to have a hard policy of no keys.
+                // Appending keys makes no real sense anyway.
+                if (col.IsKey)
+                {
+                    throw _host.Except($"Column '{sources[i]}' is key." +
+                        $"Concatenation of keys is unsupported.");
+                }
+                if (!col.ItemType.Equals(itemType))
+                {
+                    throw _host.Except($"Column '{sources[i]}' has values of {col.ItemType}" +
+                        $"which is not the same as earlier observed type of {itemType}.");
+                }
+                varVector |= col.Kind == SchemaShape.Column.VectorKind.VariableVector;
+                isNormalized &= col.IsNormalized();
+                hasCategoricals |= HasCategoricals(col);
+                hasSlotNames |= col.Kind == SchemaShape.Column.VectorKind.Scalar || col.HasSlotNames();
+            }
+            var vecKind = varVector ? SchemaShape.Column.VectorKind.VariableVector :
+                    SchemaShape.Column.VectorKind.Vector;
+
+            List<SchemaShape.Column> meta = new List<SchemaShape.Column>();
+            if (isNormalized)
+                meta.Add(new SchemaShape.Column(MetadataUtils.Kinds.IsNormalized, SchemaShape.Column.VectorKind.Scalar, BoolType.Instance, false));
+            if (hasCategoricals)
+                meta.Add(new SchemaShape.Column(MetadataUtils.Kinds.CategoricalSlotRanges, SchemaShape.Column.VectorKind.Vector, NumberType.I4, false));
+            if (hasSlotNames)
+                meta.Add(new SchemaShape.Column(MetadataUtils.Kinds.SlotNames, SchemaShape.Column.VectorKind.Vector, TextType.Instance, false));
+
+            return new SchemaShape.Column(name, vecKind, itemType, false, new SchemaShape(meta));
+        }
+
+        public SchemaShape GetOutputSchema(SchemaShape inputSchema)
+        {
+            _host.CheckValue(inputSchema, nameof(inputSchema));
+            var result = inputSchema.Columns.ToDictionary(x => x.Name);
+            result[_name] = CheckInputsAndMakeColumn(inputSchema, _name, _source);
+            return new SchemaShape(result.Values);
+        }
+    }
+
+    // REVIEW: Note that the presence of this thing is a temporary measure only.
+    // If it is cleaned up by code complete so much the better, but if not we will
+    // have to wait a little bit.
+    internal sealed class ConcatTransformer : ITransformer, ICanSaveModel
```

Comment:
This is not intended to be permanent, but is necessary for reasons of expediency. 😛 @GalOshri , @Zruty0

Reply:
Can we have issue, so this wouldn't be buried inside code? 

---
In reply to: [216526746](https://github.com/dotnet/machinelearning/pull/881#discussion_r216526746) [](ancestors = 216526746)

Reply:
I'm already working on concat transformer

---
In reply to: [216746880](https://github.com/dotnet/machinelearning/pull/881#discussion_r216746880) [](ancestors = 216746880,216526746)

Reply:
I can confirm this, @Zruty0 will clean up my mess. 😃